src/hotspot/share/gc/shenandoah/heuristics/shenandoahOldHeuristics.cpp

```diff
@@ -236,9 +236,9 @@ void ShenandoahOldHeuristics::prepare_for_old_collections() {
 
     if (percent_garbage < collection_threshold_garbage_percent) {
       _hidden_next_old_collection_candidate = 0;
-      _hidden_old_collection_candidates = i;
-      _first_coalesce_and_fill_candidate = i;
-      _old_coalesce_and_fill_candidates = cand_idx - i;
+      _hidden_old_collection_candidates = (uint)i;
+      _first_coalesce_and_fill_candidate = (uint)i;
+      _old_coalesce_and_fill_candidates = (uint)(cand_idx - i);
```

Comment:
I wonder if it would be more useful to change the type of i to uint, or maybe the fields to size_t instead?

Reply:
Maybe. That code seems to be in flux, so I would just fix the builds at this time, and then reconsider changing the types more globally.